### PR TITLE
chore: add run.Env that loads a stack run environment

### PR DESCRIPTION
**Pull request by katcipis**
This will be used on the next PR to implement the run with env feature. When we are iterating the ordered stacks this function will be used to load the stack env so we can pass it to the created command.

### CODE REVIEW

File: run/env.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+)
+
+// EnvVars represents a set of environment variables to be used
+// when running commands. Each string follows the same format used
+// on os.Environ and can be used to set env on exec.Cmd.
+type EnvVars []string
+
+// Env will load environment variables to be exported when running any command
+// inside the given stack. The order of the env vars is guaranteed to be the same
+// and is ordered lexicographically.
```
Comment (katcipis) [inline run/env.go:51]: This is not a hard requirement for setting the env on the command, but it is useful for testing + Im planning on have an experimental command to show the environment vars for each stack, that will make the output nice/deterministic there too.

File: run/env.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	"github.com/mineiros-io/terramate/errors"
+	"github.com/mineiros-io/terramate/hcl"
+	"github.com/mineiros-io/terramate/stack"
+	"github.com/rs/zerolog/log"
+	"github.com/zclconf/go-cty/cty"
+)
+
+const (
+	// ErrParsingCfg indicates that an error happened while parsing configuration.
+	ErrParsingCfg errors.Kind = "parsing run env cfg"
```
Comment (i4ki) [inline run/env.go:30]:
```suggestion
	ErrParsingCfg errors.Kind = "parsing terramate.config.run.env configuration"
```
wdyt?

File: stack/eval.go [hunk 3]
```diff
@@ -44,6 +46,17 @@ func (e *EvalCtx) SetMetadata(sm Metadata) {
 	e.evalctx.SetNamespace("terramate", metaToCtyMap(sm))
 }
 
+// SetEval sets the given environment on the env namespace of the evaluation context.
+// environ must be on the same format as os.Environ().
+func (e *EvalCtx) SetEval(environ []string) {
```
Comment (i4ki) [inline stack/eval.go:51]: SetEval? should not it be `SetEnviron()` or something?

**Review by i4ki — DISMISSED** [review]
LGTM but left some suggestions and questions.

File: stack/eval.go (hunk 3 above)
Comment (katcipis) [reply stack/eval.go:51]: The name of the namespace is "eval", so it seemed more straithforward :thinking: (maybe call SetEvalNamespace, but the other namespace setters are just SetNamespaceName).

Comment (katcipis) [reply stack/eval.go:51]: Oh my crap, it should be SetEnv :laughing: ....

**Review by i4ki — APPROVED** [review]

**Pull request merged**